msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -704,4 +709,61 @@ public IAuthenticationResult acquireTokenSilent(@NonNull final AcquireTokenSilen
                 SINGLE_ACCOUNT_PCA_ACQUIRE_TOKEN_SILENT_WITH_PARAMETERS
         );
     }
+
+    @Override
+    protected DeviceCodeFlowCommandCallback getDeviceCodeFlowCommandCallback(@NonNull final DeviceCodeFlowCallback callback) {
+        return new DeviceCodeFlowCommandCallback<LocalAuthenticationResult, BaseException>() {
+
+            @Override
+            public void onUserCodeReceived(@NonNull final String vUri,
+                                           @NonNull final String userCode,
+                                           @NonNull final String message,
+                                           @NonNull final Date sessionExpirationDate) {
+                callback.onUserCodeReceived(vUri, userCode, message, sessionExpirationDate);
+            }
+
+            @Override
+            public void onTaskCompleted(LocalAuthenticationResult tokenResult) {
+                // Convert tokenResult to an AuthenticationResult object
+                final IAuthenticationResult convertedResult = AuthenticationResultAdapter.adapt(
+                        tokenResult);
+
+                // Type cast the interface object
+                final AuthenticationResult authResult = (AuthenticationResult) convertedResult;
```

Comment:
Why do we need this cast?

Reply:
I think it's a remnant of some code I wrote during the internship, don't think it's needed to be honest.

Reply:
Ah, it's actually because onTokenReceived in the callback is expecting an AuthenticationResult, not IAuthenticationResult. I'll change the declaration in IPCA and update everything to use IAuthenticationResult